### PR TITLE
kube-proxy: do not export network programming latency for deleted Endpoints

### LINKED ISSUE #640: Network Programming Latency: investigate spikes
http://perf-dash.k8s.io/#/?jobname=gce-100Nodes&metriccategoryname=KubeProxy&metricname=NetworkProgrammingLatency

asd

/assign @oxddr

### PR DESCRIPTION
**Pull request by oxddr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
At the moment the annotation used for calculations is not set when Endpoint is deleted, causing skewed results.

Also improve logging: log name of the endpoint with its latency.

**Which issue(s) this PR fixes**:
ref https://github.com/kubernetes/kubernetes/issues/81360 
fixes https://github.com/kubernetes/perf-tests/issues/640

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
n/a
```

### CODE REVIEW

File: pkg/proxy/endpoints.go [hunk 1]
```diff
@@ -140,9 +140,12 @@ func (ect *EndpointChangeTracker) Update(previous, current *v1.Endpoints) bool {
 		change.previous = ect.endpointsToEndpointsMap(previous)
 		ect.items[namespacedName] = change
 	}
-	if t := getLastChangeTriggerTime(endpoints); !t.IsZero() {
-		ect.lastChangeTriggerTimes[namespacedName] =
-			append(ect.lastChangeTriggerTimes[namespacedName], t)
+	// TODO(#81360): ignore case when Endpoint is deleted.
+	if current != nil {
```
Comment (wojtek-t) [inline pkg/proxy/endpoints.go]: Instead of this if, I would prefer the following:
```
if t := getlastChangeTriggerTime(current); ... {
}
```
[so just change endpoints to current here]

And the in getLastChangeTriggerTime handle the nil case:
```
if endpoints == nil {
  # TODO(#81360): Figure out how to handle Endpoints deletion
  return time.Time{}
}
```

File: pkg/proxy/endpoints.go [hunk 2]
```diff
@@ -197,14 +200,14 @@ type UpdateEndpointMapResult struct {
 	StaleServiceNames []ServicePortName
 	// List of the trigger times for all endpoints objects that changed. It's used to export the
 	// network programming latency.
-	LastChangeTriggerTimes []time.Time
+	LastChangeTriggerTimes map[types.NamespacedName][]time.Time
```
Comment (wojtek-t) [inline pkg/proxy/endpoints.go]: I don't understand why this is needed?
[or at the very least, it doesn't seem to be related to handling deletedEndpoints, so please remove it from this PR]

Comment (oxddr) [reply pkg/proxy/endpoints.go]: This change is needed to improve logging. So far we've only logged latency, but not the name of the endpoint, making the problem hard to debug. 

I'm fine with moving this to a separate PR - please let me know.

Comment (wojtek-t) [reply pkg/proxy/endpoints.go]: Yes - please move it to a separate PR - please don't mix two separate things in a single PR.
